internal/services/healthcare/healthcare_fhir_resource_test.go

```diff
@@ -0,0 +1,384 @@
+	r := HealthcareApiFhirServiceResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.updateCors(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccHealthcareApiFhirService_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_healthcare_fhir_service", "test")
+	r := HealthcareApiFhirServiceResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func (HealthcareApiFhirServiceResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := parse.FhirServiceID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := clients.HealthCare.HealthcareWorkspaceFhirServiceClient.Get(ctx, id.ResourceGroup, id.WorkspaceName, id.Name)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving Healthcare api fhir service %s: %+v", *id, err)
+	}
+	return utils.Bool(resp.FhirServiceProperties != nil), nil
+}
+
+func (r HealthcareApiFhirServiceResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_healthcare_fhir_service" "test" {
+  name                = "fhir%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  workspace_id        = azurerm_healthcare_workspace.test.id
+  kind                = "fhir-R4"
+  authentication {
+    authority = "https://login.microsoftonline.com/72f988bf-86f1-41af-91ab-2d7cd011db47"
+    audience  = "https://acctestfhir.fhir.azurehealthcareapis.com"
+  }
+  depends_on = [azurerm_healthcare_workspace.test]
+}
+`, r.template(data), data.RandomInteger)
+}
+
+func (r HealthcareApiFhirServiceResource) updateIdentity(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+resource "azurerm_healthcare_fhir_service" "test" {
+  name                = "fhir%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  workspace_id        = azurerm_healthcare_workspace.test.id
+  kind                = "fhir-R4"
+  authentication {
+    authority = "https://login.microsoftonline.com/72f988bf-86f1-41af-91ab-2d7cd011db47"
+    audience  = "https://acctestfhir.fhir.azurehealthcareapis.com"
+  }
+  identity {
+    type = "SystemAssigned"
+  }
+  depends_on = [azurerm_healthcare_workspace.test]
```

Comment:
could we format this terraform betteR?
```suggestion
  kind                = "fhir-R4"
  
  authentication {
    authority = "https://login.microsoftonline.com/72f988bf-86f1-41af-91ab-2d7cd011db47"
    audience  = "https://acctestfhir.fhir.azurehealthcareapis.com"
  }
  
  identity {
    type = "SystemAssigned"
  }
  
  depends_on = [azurerm_healthcare_workspace.test]
```